packages/@aws-cdk/aws-cloudfront/lib/cache-policy.ts

```diff
@@ -0,0 +1,298 @@
+  /** Designed for use with an origin that is an AWS Elemental MediaPackage endpoint. */
+  public static readonly ELEMENTAL_MEDIA_PACKAGE = CachePolicy.fromManagedCachePolicy('08627262-05a9-4f76-9ded-b50ca2e3a84f');
+
+  /** Imports a Cache Policy from its id. */
+  public static fromCachePolicyId(scope: Construct, id: string, cachePolicyId: string): ICachePolicy {
+    return this.fromCachePolicyAttributes(scope, id, { cachePolicyId });
+  }
+
+  /** Imports a Cache Policy from its attributes. */
+  public static fromCachePolicyAttributes(scope: Construct, id: string, attrs: CachePolicyAttributes): ICachePolicy {
+    return new class extends Resource implements ICachePolicy {
+      public readonly cachePolicyId = attrs.cachePolicyId;
+    }(scope, id);
+  }
+
+  /** Use an existing managed cache policy. */
+  private static fromManagedCachePolicy(managedCachePolicyId: string): ICachePolicy {
+    return new class implements ICachePolicy {
+      public readonly cachePolicyId = managedCachePolicyId;
+    }();
+  }
+
+  public readonly cachePolicyId: string;
+
+  constructor(scope: Construct, id: string, props: CachePolicyProps) {
+    super(scope, id, {
+      physicalName: props.cachePolicyName,
+    });
+
+    const minTtl = (props.minTtl ?? Duration.seconds(0)).toSeconds();
+    const defaultTtl = Math.max((props.defaultTtl ?? Duration.days(1)).toSeconds(), minTtl);
+    const maxTtl = Math.max((props.maxTtl ?? Duration.days(365)).toSeconds(), defaultTtl);
+
+    const resource = new CfnCachePolicy(this, 'Default', {
+      cachePolicyConfig: {
+        name: props.cachePolicyName,
+        comment: props.comment,
+        minTtl,
+        maxTtl,
+        defaultTtl,
+        parametersInCacheKeyAndForwardedToOrigin: this.renderCacheKey(props),
+      },
+    });
+
+    this.cachePolicyId = resource.ref;
+  }
+
+  private renderCacheKey(props: CachePolicyProps): CfnCachePolicy.ParametersInCacheKeyAndForwardedToOriginProperty {
+    const cookies = props.cookieBehavior ?? CookieBehavior.none();
+    const headers = props.headerBehavior ?? HeaderBehavior.none();
+    const queryStrings = props.queryStringBehavior ?? QueryStringBehavior.none();
+
+    return {
+      cookiesConfig: {
+        cookieBehavior: cookies.behavior,
+        cookies: cookies.cookies,
+      },
+      headersConfig: {
+        headerBehavior: headers.behavior,
+        headers: headers.headers,
+      },
+      enableAcceptEncodingGzip: props.enableAcceptEncodingGzip ?? false,
+      queryStringsConfig: {
+        queryStringBehavior: queryStrings.behavior,
+        queryStrings: queryStrings.queryStrings,
+      },
+    };
+  }
+}
+
+/**
+ * Determines whether any cookies in viewer requests are included in the cache key and
+ * automatically included in requests that CloudFront sends to the origin.
+ * @experimental
+ */
+export class CookieBehavior {
```

Comment:
I wonder whether this name is too generic. How about `CacheCookieBehavior`?

Reply:
Sure, I debated that myself. And taking a closer look at the Origin Request Policies, it looks like we'll be creating `OriginRequestCookieBehavior` and friends that are extremely similar to these. Worth having the different names.